docs/LED_Effect.md

```diff
@@ -108,38 +108,74 @@ layers:
+data, while active effects return color data, that is summed up for each LED
 they run on.
 
 #### Activating and deactivating effects
-Our example effect can be activated by running the GCode command 
-`SET_LED_EFFECT EFFECT=panel_idle`. To stop all effects which are currently 
-running on the LEDs the new effect is using, set the `REPLACE` parameter to 1: 
+Our example effect can be activated by running the GCode command
+`SET_LED_EFFECT EFFECT=panel_idle`. To stop all effects which are currently
+running on the LEDs the new effect is using, set the `REPLACE` parameter to 1:
 `SET_LED_EFFECT EFFECT=panel_idle REPLACE=1`
-Running the command `SET_LED_EFFECT EFFECT=panel_idle STOP=1` deactivates this 
-particular effect again. 
+Running the command `SET_LED_EFFECT EFFECT=panel_idle STOP=1` deactivates this
+particular effect again.
 To deactivate all effects we can use the GCode command `STOP_LED_EFFECTS`.
 To only deactivate effects for certain LEDs we can specify the LEDS parameter:
-`STOP_LED_EFFECTS LEDS="neopixel:panel_ring"` You can also specify indeces (see 
-below): `STOP_LED_EFFECTS LEDS="neopixel:panel_ring (1-7)"`. Only one 
-LED parameter can be specified at a time. To stop the effects for multiple LEDs 
+`STOP_LED_EFFECTS LEDS="neopixel:panel_ring"` You can also specify indeces (see
+below): `STOP_LED_EFFECTS LEDS="neopixel:panel_ring (1-7)"`. Only one
+LED parameter can be specified at a time. To stop the effects for multiple LEDs
 we have to run the command multiple times.
 
 #### Fading in and out
 Effects can be faded in and out by specifying the `FADETIME` parameter:
-`SET_LED_EFFECT EFFECT=panel_idle FADETIME=1.0` fades the effect in during one 
-second. Running `SET_LED_EFFECT EFFECT=panel_idle STOP=1 FADETIME=1.0` fades it 
-out in one second. We can also fade out all effects by running 
-`STOP_LED_EFFECTS FADETIME=1.0`. It is also possible to crossfade effects by 
-using the `REPLACE` parameter with `SET_LED_EFFECT` (see above): 
+`SET_LED_EFFECT EFFECT=panel_idle FADETIME=1.0` fades the effect in during one
+second. Running `SET_LED_EFFECT EFFECT=panel_idle STOP=1 FADETIME=1.0` fades it
+out in one second. We can also fade out all effects by running
+`STOP_LED_EFFECTS FADETIME=1.0`. It is also possible to crossfade effects by
+using the `REPLACE` parameter with `SET_LED_EFFECT` (see above):
 `SET_LED_EFFECT EFFECT=panel_idle REPLACE=1 FADETIME=1.0`
 
 #### Restarting Effects
-When an effect is stopped and then started again, it resumes from the frame where 
-it last left off.  To restart the effect from the beginning, specify the `RESTART` 
+When an effect is stopped and then started again, it resumes from the frame where
+it last left off.  To restart the effect from the beginning, specify the `RESTART`
 parameter: `SET_LED_EFFECT EFFECT=panel_idle RESTART=1`.
 
+#### Template processing
+The effect layers are processed as templates. That means that they can contain
```

Comment:
Maybe add a link to https://www.klipper3d.org/Command_Templates.html#template-expansion ?

Reply:
Done.